common/sagas/bity.js

```diff
@@ -1,23 +1,45 @@
 // @flow
 
-import { takeLatest, call, apply, put, select, fork } from 'redux-saga/effects';
+import { call, put, fork, take, cancel, cancelled } from 'redux-saga/effects';
+
 import type { Effect } from 'redux-saga/effects';
 import { delay } from 'redux-saga';
 import { updateBityRatesSwap } from 'actions/swap';
-import { SWAP_LOAD_BITY_RATES } from 'actions/swapConstants';
+import {
+  SWAP_LOAD_BITY_RATES,
+  SWAP_STOP_LOAD_BITY_RATES
+} from 'actions/swapConstants';
 import type { UnlockPrivateKeyAction } from 'actions/wallet';
 import { getAllRates } from 'api/bity';
 
 export function* loadBityRates(action?: any): Generator<Effect, void, any> {
-  if (!action) return;
-  // eslint-disable-next-line
-  while (true) {
-    const data = yield call(getAllRates);
-    yield put(updateBityRatesSwap(data));
-    yield call(delay, 5000);
+  try {
+    while (true) {
+      // TODO - yield put(actions.requestStart()) if we want to display swap refresh status
+      // network request
+      const data = yield call(getAllRates);
+      // action
+      yield put(updateBityRatesSwap(data));
+      // wait 5 seconds before refreshing rates
+      yield call(delay, 5000);
+    }
+  } finally {
+    if (yield cancelled()) {
+      // TODO - implement request cancel if needed
+      // yield put(actions.requestFailure('Request cancelled!'))
+    }
   }
 }
 
 export default function* bitySaga(): Generator<Effect, void, any> {
-  yield takeLatest(SWAP_LOAD_BITY_RATES, loadBityRates);
+  while (yield take(SWAP_LOAD_BITY_RATES)) {
```

Comment:
@crptm I'm not confident about this, used the pattern described here: https://redux-saga.js.org/docs/advanced/TaskCancellation.html